### PR TITLE
Add Jekyll Caching Plugin

### PR DESCRIPTION
**Pull request by Pandapip1**
Rationale: The Jekyll build and HTML check is the rate-limiting factor of the CI.

~~Blocked by #5236, as I have no way of locally testing these changes.~~

Never mind, forgot I could just point github pages to another branch. Here is the result: https://pandapip1.github.io/EIPs/

The build takes about ~25s faster.

### CODE REVIEW

**Comment by Pandapip1** [comment]
😓 that was close

File: Gemfile.lock [hunk 1]
```diff
@@ -1,274 +0,0 @@
-GEM
```
Comment (SamWilsn) [inline Gemfile.lock:1]: Why remove the lockfile?

Comment (Pandapip1) [reply Gemfile.lock:1]: The alternative is that I add something to the CI to keep it up to date. Would that be preferable?

Comment (SamWilsn) [reply Gemfile.lock:1]: Something like dependabot? Yeah, I'd be okay with that, as long as a human has to approve the change.

**Comment by Pandapip1** [comment]
Let's try this again, but this time, the CI makes the gemfile.

**Comment by Pandapip1** [comment]
I give up.

**Pull request closed without merging**